### PR TITLE
Adjust shared subscription sample to better fit docs

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Makes a few minor changes to better fit the AWS docs for shared subscriptions. Also adds more logging and removes some temporary code added for compatibility prior to shared subscription support public release.

_____________

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samples/browser/shared_subscription/README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ Replace with the following with the data from your AWS account:
 * `<region>`: The AWS IoT Core region where you created your AWS IoT Core thing you wish to use with this sample. For example `us-east-1`.
 * `<account>`: Your AWS IoT Core account ID. This is the set of numbers in the top right next to your AWS account name when using the AWS IoT Core website.
 
-Note that in a real application, you may want to avoid the use of wildcards in your ClientID or use them selectively. Please follow best practices when working with AWS on production applications using the SDK. Also, for the purposes of this sample, please make sure your policy allows a client ID of `test-*` to connect or use `--client_id <client ID here>` to send the client ID your policy supports.
+Note that in a real application, you may want to avoid the use of wildcards in your ClientID and shared subscription group names/identifiers. Wildcards should be used only selectively. Please follow best practices when working with AWS on production applications using the SDK. Also, for the purposes of this sample, please make sure your policy allows a client ID of `test-*` to connect or use `--client_id <client ID here>` to send the client ID your policy supports.
```
Comment (sbSteveK) [inline samples/browser/shared_subscription/README.md:74]: Trivial:
"Wildcards should be used only selectively." => "Wildcards should only be used selectively." ?

**Review by sbSteveK — APPROVED** [review]

File: samples/browser/shared_subscription/README.md (hunk 1 above)
Comment (sbSteveK) [reply samples/browser/shared_subscription/README.md:74]: This description is the same across all 4 prs. I guess if changed here, maybe change elsewhere as well. But again, super trivial and just personal preference =P

Comment (TwistedTwigleg) [reply samples/browser/shared_subscription/README.md:74]: Yeah, that is better. I will adjust all the PRs accordingly 👍

**Comment by TwistedTwigleg** [comment]
Thanks! Merging into `main`...

**Pull request merged**